### PR TITLE
Integrate AMD GPU in CI/CD environment

### PR DESCRIPTION
**Pull request by mfuntowicz**

### CODE REVIEW

File: .github/workflows/self-push.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@ on:
     types: [completed]
```
Comment (ydshieh) [inline .github/workflows/self-push.yml:7]: I recommend to have `self-push-amd.yml` which is copied from `self-push.yml` but with the modification to use AMD related stuff.

So the Nvidia/AMD jobs won't be mixed in a single file. More advantages:

- The jobs will be clear to navigate (no nvidia/amd jobs being all in the same run page)
- We don't need to rework the Slack report script `utils/notification_service.py` (which is already too complex IMO)
  - we don't need to rename `machine_type`  (and the tags) to `single-amdgpu` / `multi-amdgpu` (this requires rework on ``utils/notification_service.py` to make the report working
  - we only need to tag `docker-amdgpu` and `single-gpu` (or `multi-gpu`) on the runner, and use `machine_type: [ single-gpu ]` (or `machine_type: [ mult-amdgpu ]`)

Comment (ydshieh) [reply .github/workflows/self-push.yml:7]: Then both nvidia/amd workflow will be triggered (at the same time) independently, see


[.github/workflows/self-push-caller.yml](https://github.com/huggingface/transformers/blob/main/.github/workflows/self-push-caller.yml)

**Comment by mfuntowicz** [comment]
@ydshieh @LysandreJik I think we are in a good shape for review and merging.

What we did: 
- [x] Added custom runners with tags `docker-amdgpu`,  `single-amdgpu`, `multi-amdgpu`, `mi210`
- [x] Provide a custom PyTorch GPU Dockerfile for AMD dependencies
- [x] Create a new `self-push-amd.yml` workflow file for everything related to AMD testing
- [x] Validated the workflow against a simple BERT modifications 

What we cannot ensure as of today:
- [ ] All the current tests being executed on main will be green 😅

File: docker/transformers-pytorch-gpu/Dockerfile.amd [hunk 2]
```diff
@@ -0,0 +1,29 @@
+FROM rocm/pytorch:rocm5.6_ubuntu20.04_py3.8_pytorch_2.0.1
```
[the remark this answers is not in the document]
Comment (mfuntowicz) [reply docker/transformers-pytorch-gpu/Dockerfile.amd:1]: This image is built and published by AMD directly on the Docker Hub.

I can add a job in the `build-docker-images.yml` workflow to build `transformers-pytorch-latest-amdgpu-push-ci`. 
It has been pushed to our Docker Hub organisation by myself 🙂

**Comment by ydshieh** [comment]
Hi @mfuntowicz 

Looking at the runs in https://github.com/huggingface/transformers/actions/workflows/self-push-amd.yml, you will see no test job (Model test) is being triggered (as no test is being collected).

Also the slack report won't work as the tag is sitll using `single-amdgpu` instead of `single-gpu`.

**Review by ydshieh — APPROVED** [review]
🔥

**Comment by ydshieh** [comment]
@LysandreJik in case you want to take a final look :-)

**Comment by ydshieh** [comment]
Merge now so @mfuntowicz can show progress to AMD team today.

**Review by LysandreJik — APPROVED** [review]
Ok LGTM

**Pull request merged**